src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -903,9 +903,9 @@ trait Infer extends Checkable {
         isAsSpecificValueType(rtpe1, tpe2, undef1 ::: tparams1, undef2)
       case _                         =>
         tpe2 match {
-          case PolyType(tparams2, rtpe2) => isAsSpecificValueType(tpe1, rtpe2, undef1, undef2 ::: tparams2)
-          case _ if !currentRun.isScala3 => existentialAbstraction(undef1, tpe1) <:< existentialAbstraction(undef2, tpe2)
-          case _                         =>
+          case PolyType(tparams2, rtpe2)                   => isAsSpecificValueType(tpe1, rtpe2, undef1, undef2 ::: tparams2)
+          case _ if !currentRun.isScala3ImplicitResolution => existentialAbstraction(undef1, tpe1) <:< existentialAbstraction(undef2, tpe2)
```

Comment:
This incurs the implicit conversion. I wonder why I don't see a warning currently.

Reply:
I saw your comment and I was going to quickly make a change to avoid the conversion but upon checking out the most recent code the conversion doesn't happen anymore with your change https://github.com/scala/scala/commit/0d252f6b1f6211a56030bfd1bdcf509a7ad084f8#diff-7b7600384a160cb93e5eceff814f6aed9f5c7b72b752a18c3f61d895482e75fcR907

Reply:
Thanks for checking my check. (That's what I meant by I caught it in my PR, which was about the same time as yours.) I might change them to vars, except there were a lot of `isScala3` calls, so removing the `.value` is not appealing. But that is why `sed` is the "scala editor".